src/nvim/path.c

```diff
@@ -1779,6 +1793,10 @@ int path_with_url(const char *fname)
     return 0;
   }
 
+  if (path_has_drive_letter(fname)) {
```

Comment:
Shouldn't have we ifdef'd guarded this for windows?

Reply:
If a buffer has `c:/foo/bar` and user hits `gx` (which I assume hits this codepath), we wouldn't want to treat it as a URL. Other benefits: (1) nice to have the same codepaths on all platforms if possible, and (2) makes it easier to get test coverage (because unit tests don't run on Windows).